src/Xrefcheck/Scanners/Markdown/ErrorMessage.hs

```diff
@@ -0,0 +1,51 @@
+{- SPDX-FileCopyrightText: 2018-2019 Serokell <https://serokell.io>
+ -
+ - SPDX-License-Identifier: MPL-2.0
+ -}
+
+module Xrefcheck.Scanners.Markdown.ErrorMessage
+  ( makeError
+  ) where
+
+import Universum
+
+import Xrefcheck.Scanners.Markdown.IgnoreMode (IgnoreMode (..))
+import Xrefcheck.Scanners.Markdown.Node (PosInfo, prettyPos)
+
+fileMsg :: Text
+fileMsg =
+  "\"ignore file\" must be at the top of \
+  \markdown or right after comments at the top"
+
+linkMsg :: Text
+linkMsg = "expected a LINK after \"ignore link\" "
+
+paragraphMsg :: Text -> Text
+paragraphMsg txt = unwords
+  [ "expected a PARAGRAPH after \
+      \\"ignore paragraph\", but found"
+  , txt
+  , ""
+  ]
+
+unrecognisedMsg :: Text -> Text
+unrecognisedMsg txt = unwords
+  [ "unrecognised option"
+  , "\"" <> txt <> "\""
+  , "perhaps you meant \
+      \<\"ignore link\"|\"ignore paragraph\"|\"ignore file\"> "
+  ]
```

Comment:
I think this is not a good thing actually. Putting error messages far from the use site, we encourage their divergence from  the actual problem (e.g. the actual error can obtain one more condition when it fires, but it is now easy to forget to update the error message respectively).

I think error messages and where they are used should fit into screen, maybe but in `where`/`letl` block if found to make the code less readable.